src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -5700,27 +5700,47 @@ def _create_run(
 
         # Add logs entry for the run if exists
         if pipeline_run.logs is not None:
-            self._get_reference_schema_by_id(
-                resource=pipeline_run,
-                reference_schema=StackComponentSchema,
-                reference_id=pipeline_run.logs.artifact_store_id,
-                session=session,
-                reference_type="logs artifact store",
-            )
+            with session.no_autoflush:
+                self._get_reference_schema_by_id(
+                    resource=pipeline_run,
+                    reference_schema=StackComponentSchema,
+                    reference_id=pipeline_run.logs.artifact_store_id,
+                    session=session,
+                    reference_type="logs artifact store",
+                )
 
-            log_entry = LogsSchema(
-                uri=pipeline_run.logs.uri,
-                pipeline_run_id=new_run.id,
-                artifact_store_id=pipeline_run.logs.artifact_store_id,
-            )
-            session.add(log_entry)
+                log_entry = LogsSchema(
+                    uri=pipeline_run.logs.uri,
+                    pipeline_run_id=new_run.id,
+                    artifact_store_id=pipeline_run.logs.artifact_store_id,
+                )
+                session.add(log_entry)
 
         try:
             session.commit()
-        except IntegrityError:
+        except IntegrityError as e:
             # We have to rollback the failed session first in order to
             # continue using it
             session.rollback()
+
+            # Check if this is a duplicate run name error
```

Comment:
There is a better place to detect this: This `_create_run` method is called from `get_or_create_run`, and if you scroll right to the bottom of that method you can find an explanation where it mentions name duplicates. If that works, we can avoid this ugly check for the constraint in the error message